bccsp/pkcs11/pkcs11.go

```diff
@@ -84,25 +84,44 @@ func loadLib(lib, pin, label string) (*pkcs11.Ctx, uint, *pkcs11.SessionHandle,
 func (csp *impl) getSession() (session pkcs11.SessionHandle) {
 	select {
 	case session = <-csp.sessions:
-		logger.Debugf("Reusing existing pkcs11 session %+v on slot %d\n", session, csp.slot)
+		_, err := csp.ctx.GetSessionInfo(session)
```

Comment:
So, we failed to get the session info and assume that means the session is bad. I'll buy that.

However, if we think it's bad, why aren't we attempting to close it to ensure that any resources being held on the cgo side of this are getting released?

Reply:
I agree, I will add CloseSession before creating a new session.